src/main.rs

```diff
@@ -77,113 +80,117 @@ fn main() {
-                .long("start-time-us")
-                .takes_value(true)
-                .hidden(true),
-        )
-        .arg(
-            Arg::with_name("start-time-cpu-us")
-                .long("start-time-cpu-us")
-                .takes_value(true)
-                .hidden(true),
-        )
-        .arg(
-            Arg::with_name("config-file")
-                .long("config-file")
-                .help("Path to a file that contains the microVM configuration in JSON format.")
-                .takes_value(true)
-                .required(false),
-        )
-        .arg(
-            Arg::with_name("no-api")
-                .long("no-api")
-                .help("Optional parameter which allows starting and using a microVM without an active API socket.")
-                .takes_value(false)
-                .required(false)
-                .requires("config-file")
-        )
-        .get_matches();
-
-    let bind_path = cmd_arguments
-        .value_of("api_sock")
+    let mut args: Vec<String> = env::args().collect();
+    let mut args_parser = ArgParser::default();
+
+    args_parser.arg(
+        ArgumentInfo::new("api-sock")
+            .takes_value(true)
+            .default_value(DEFAULT_API_SOCK_PATH)
+            .help("Path to unix domain socket used by the API"),
+    );
+    args_parser.arg(
+        ArgumentInfo::new("id")
+            .takes_value(true)
+            .default_value(DEFAULT_INSTANCE_ID)
+            .help("MicroVM unique identifier"),
+    );
+    args_parser.arg(
+        ArgumentInfo::new("seccomp-level")
+            .takes_value(true)
+            .default_value("2")
+            .help(
+                "Level of seccomp filtering that will be passed to executed path as \
+                argument.\n
+                    - Level 0: No filtering.\n
+                    - Level 1: Seccomp filtering by syscall number.\n
+                    - Level 2: Seccomp filtering by syscall number and argument values.\n
+                ",
+            ),
+    );
+    args_parser.arg(
+        ArgumentInfo::new("start-time-us")
+            .takes_value(true)
+            .optional(true),
+    );
+    args_parser.arg(
+        ArgumentInfo::new("start-time-cpu-us")
+            .takes_value(true)
+            .optional(true),
+    );
+    args_parser.arg(
+        ArgumentInfo::new("config-file")
+            .takes_value(true)
+            .optional(true)
+            .help("Path to a file that contains the microVM configuration in JSON format."),
+    );
+    args_parser.arg(ArgumentInfo::new(
+        "no-api")
+        .takes_value(false)
+        .requires("config-file")
+        .optional(true)
+        .help("Optional parameter which allows starting and using a microVM without an active API socket.")
+    );
+
+    // Remove the first element of `args` as it should be the binary name.
+    args.remove(0);
+    args_parser.set_matches(&args).unwrap_or_else(|err| {
```

Comment:
I think it would be cleaner if `set_matches` would do `env::args().collect();` inside (just like clap `get_matches`), instead of receiving the arguments when called. I would also propose to rename this method to something like `parse_cmd_line()` or `parse()`, etc.

Reply:
I moved `env::args().collect()` in ArgParser implementation, but in a separate function in order to be able to test `set_maches` (which I've also renamed to `get_matches`).